drivers/pinctrl/pinctrl_mchp_xec.c

```diff
@@ -17,51 +17,80 @@
  * slew rate and driver strength in the second control register.
  * We compute the register index from the beginning of the GPIO
  * control address space which is the same range of the PINCTRL
- * parent node.
+ * parent node. A zero value in the PINCTRL pinmux field means
+ * do not touch.
  */
-
 static void config_drive_slew(struct gpio_regs * const regs, uint32_t idx, uint32_t conf)
 {
-	uint32_t slew = conf & (MCHP_XEC_OSPEEDR_MASK << MCHP_XEC_OSPEEDR_POS);
-	uint32_t drvstr = conf & (MCHP_XEC_ODRVSTR_MASK << MCHP_XEC_ODRVSTR_POS);
-	uint32_t val = 0;
-	uint32_t mask = 0;
-
-	if (slew != MCHP_XEC_OSPEEDR_NO_CHG) {
-		mask |= MCHP_GPIO_CTRL2_SLEW_MASK;
-		if (slew == MCHP_XEC_OSPEEDR_FAST) {
+	uint32_t slew = (conf >> MCHP_XEC_SLEW_RATE_POS) & MCHP_XEC_SLEW_RATE_MSK0;
+	uint32_t drvstr = (conf >> MCHP_XEC_DRV_STR_POS) & MCHP_XEC_DRV_STR_MSK0;
+	uint32_t msk = 0, val = 0;
+
+	if (slew) {
+		msk |= MCHP_GPIO_CTRL2_SLEW_MASK;
+		/* slow slew value is 0 */
+		if (slew == MCHP_XEC_SLEW_RATE_FAST0) {
 			val |= MCHP_GPIO_CTRL2_SLEW_FAST;
 		}
 	}
-	if (drvstr != MCHP_XEC_ODRVSTR_NO_CHG) {
-		mask |= MCHP_GPIO_CTRL2_DRV_STR_MASK;
-		val |= (drvstr << MCHP_GPIO_CTRL2_DRV_STR_POS);
+
+	if (drvstr) {
+		msk |= MCHP_GPIO_CTRL2_DRV_STR_MASK;
+		/* drive strength values are 0 based */
+		val |= ((drvstr - 1u) << MCHP_GPIO_CTRL2_DRV_STR_POS);
 	}
 
-	if (!mask) {
+	if (!msk) {
 		return;
 	}
 
-	regs->CTRL2[idx] = (regs->CTRL2[idx] & ~mask) | (val & mask);
+	regs->CTRL2[idx] = (regs->CTRL2[idx] & ~msk) | (val & msk);
+}
+
+/* Internal pulls feature:
```

Comment:
Conventionally, block comments in Zephyr usually keep `/*` and `*/` on a separate lines.

https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style
https://kernel.org/doc/html/latest/process/coding-style.html#commenting

Reply:
Updated the block comments as per coding-style.